apps/gitlabpipelinestatus/gitlab_pipeline.star

```diff
@@ -0,0 +1,80 @@
+canceled = base64.decode("""iVBORw0KGgoAAAANSUhEUgAAACEAAAAgCAYAAACcuBHKAAABEUlEQVRYR+2XwRnCIAyFYRmH8OLRNZzDAZzDNTx6cQiXoYoFk34BXvhM5dBem7Y/Ly8P6t0Alx+AwUWIcD2HFow/XcyAPQKQAK1AvhD7Q1mMxz3eM4eofSCpZQ7R8sQqSvwVgk3Hyxd+d8xTYN2GbPgNYpaCh5WiHVK+9E5PF0R43oKbs4MZupY1adXEcypP1CbnvXpN6krpW1UC2Vd6IWjuNCEkFWg7KATqieXofyBSjxfGLLXBBILKjqzGHAKJb1oDeaIx+uyg0uty5DmqsOgJ7epLOSG1Utp/fgJRGt3VIVD1TJXYILQKIPVm/xLI2LJdFKHV1iBHw1yjfTlaP4QSyDEgbedmnkAViyCaYqvaCbHH/oEngJdpAAAAAElFTkSuQmCC""")
+manual = base64.decode("""iVBORw0KGgoAAAANSUhEUgAAACEAAAAgCAYAAACcuBHKAAABBUlEQVRYR+2X4RGDIAyFYY2O1jkco3N0tK6BVkSDF+CF6+txnv416peXlwS9G+DyAzC4FSK8XGjB+CnGMi6PAKQPs0AOiEclx0+8R4eofSCpRYdAan1tiKw7Fl/459EF7DLshr8hNinyYWUohzZfeo3bBRHey4TdZkfWVbVZk7IWnjN5ota+3+wtU1ebvlUlkL3SCyEncBNCU0GWQ0Kgnji3foRINT4Zs1QGCoSUHcmGDoHsEBkDeaLR+tlBpdflyHNSYdUT1uxLc0IrpbZ/fgJRat2/Q6DqUZW4IawKIPG0fwmkbbMtitBaY5Cj4R5jfTkaP4QSyDEgrXOaJ1DFVhBLMCt2BhiwvEkYq+zVAAAAAElFTkSuQmCC""")
+pipeline_dict = dict([("invalid", [failed, "API call failed!"]), ("failed", [failed, "Build failed!"]), ("success", [success, "Success"]), ("running", [running, "Running"]), ("created", [createdPendingSkipped, "Created"]), ("pending", [createdPendingSkipped, "Pending"]), ("skipped", [createdPendingSkipped, "Skipped"]), ("canceled", [canceled, "Canceled"]), ("manual", [manual, "Set to manual"])])
+
+def main(config):
+    token = config.get("api-token")
+    projectId = config.get("project-id")
+    branch = config.get("branch")
+    status = get_pipeline_status(token, projectId, branch)
+    ICON = render.Image(src = pipeline_dict.get(status)[0])
+
+    box = render.Column(
+        children = [
+            render.Box(width = 1, height = 12, color = "#000000"),
+            render.WrappedText(content = pipeline_dict.get(status)[1], font = "tom-thumb"),
+        ],
+    )
+    return render.Root(
+        child = render.Row(
+            children =
+                [ICON, box],
+        ),
+    )
+
+def get_pipeline_status(accesstoken, id, ref):
+    # Set the GitLab API endpoint and access token
+    api_endpoint = "https://gitlab.com/api/v4"
+
+    # Specify the project and branch for which to check the pipeline status
+    pipeline_url = "%s/projects/%s/pipelines?ref=%s&access_token=%s" % (api_endpoint, id, ref, accesstoken)
+    pipeline_data = http.get(pipeline_url)
+    if pipeline_data.status_code != 200:
+        return "invalid"
+    pipeline_data = pipeline_data.json()
+    most_recent_pipeline = pipeline_data[0]
+    return most_recent_pipeline["status"]
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "api-token",
+                name = "Your Gitlab access token",
+                desc = "Your Gitlab access token",
+                icon = "gear",
+                default = "glpat-qGfEe9MNpEu-SzFp3sBY",
```

Comment:
If this is a real API token, you will probably want to revoke it on Gitlab as it's now in the open. The `secret.star` module allows you to include secret data in apps.